workflow-controller/controller-service/src/main/java/io/hyscale/controller/util/LoggerUtility.java

```diff
@@ -121,6 +124,70 @@ public void deploymentLogs(WorkflowContext context) throws HyscaleException {
             WorkflowLogger.footer();
         }
     }
+    
+    /**
+     * implementation :
+     * <b>
+     * 1. Replica name provided by user, validate and return replica name
+     * 2. Replica name not provided by user
+     *      a. If single replica, return replica name
+     *      b. else prompt user to provide replica name
+     * </b>
+     * @param authConfig
+     * @param appName
+     * @param serviceName
+     * @param replicaName
+     * @param namespace
+     * @return replica name
+     * @throws HyscaleException
+     */
+    private String getPodName(AuthConfig authConfig, String appName, String serviceName, String replicaName,
+            String namespace) throws HyscaleException {
+        List<V1Pod> existingPods = null;
+        try {
```

Comment:
You are validating replica name from the exiting pods .
getPodName does not sound good to this functionality , rename it to validateReplicaName...based on the output fetch the logs from deployer directly. Keep it simple